outlines/caching.py

```diff
@@ -1,14 +1,18 @@
 import os
 from typing import Callable, Optional
-
+import hashlib
+import cloudpickle
+import torch
 from perscache import Cache, NoCache
 from perscache.serializers import JSONSerializer
 from perscache.storage import LocalFileStorage
+from diskcache import Cache as DiskCache
 
 home_dir = os.path.expanduser("~")
 cache_dir = os.environ.get("OUTLINES_CACHE_DIR", f"{home_dir}/.cache/outlines")
 memory = Cache(serializer=JSONSerializer(), storage=LocalFileStorage(cache_dir))
-
+diskcache_memory = DiskCache(cache_dir, eviction_policy = 'none', cull_limit=0)
+_discache_enabled = True
```

Comment:
`diskcache_caching_enabled`? Is there a similar class as in perscache to replace the memory with?

Reply:
changed the name.